fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/PushdownFilterThroughWindow.java

```diff
@@ -81,88 +55,31 @@ public Rule build() {
-                return filter;
-            }
-
-            // Check the filter conditions. Now, we currently only support simple conditions of the form
-            // 'column </ <=/ = constant'. We will extract some related conjuncts and do some check.
-            Set<Expression> conjuncts = filter.getConjuncts();
-            Set<Expression> relatedConjuncts = extractRelatedConjuncts(conjuncts, windowExpr.getExprId());
-
-            boolean hasPartitionLimit = false;
-            long partitionLimit = Long.MAX_VALUE;
-
-            for (Expression conjunct : relatedConjuncts) {
-                Preconditions.checkArgument(conjunct instanceof BinaryOperator);
-                BinaryOperator op = (BinaryOperator) conjunct;
-                Expression leftChild = op.children().get(0);
-                Expression rightChild = op.children().get(1);
-
-                Preconditions.checkArgument(leftChild instanceof SlotReference
-                        && rightChild instanceof IntegerLikeLiteral);
-
-                long limitVal = ((IntegerLikeLiteral) rightChild).getLongValue();
-                // Adjust the value for 'limitVal' based on the comparison operators.
-                if (conjunct instanceof LessThan) {
-                    limitVal--;
+            Set<Expression> commonPartitionKeys = window.getCommonPartitionKeyFromWindowExpressions();
+            Set<Expression> bottomConjuncts = Sets.newHashSet();
+            Set<Expression> upperConjuncts = Sets.newHashSet();
+            for (Expression expr : filter.getConjuncts()) {
+                boolean pushed = false;
+                for (Expression partitionKey : commonPartitionKeys) {
+                    if (partitionKey.getInputSlots().containsAll(expr.getInputSlots())) {
+                        bottomConjuncts.add(expr);
+                        pushed = true;
+                        break;
+                    }
                 }
-                if (limitVal < 0) {
-                    return new LogicalEmptyRelation(ctx.statementContext.getNextRelationId(), filter.getOutput());
-                }
-                if (hasPartitionLimit) {
-                    partitionLimit = Math.min(partitionLimit, limitVal);
-                } else {
-                    partitionLimit = limitVal;
-                    hasPartitionLimit = true;
+                if (!pushed) {
+                    upperConjuncts.add(expr);
                 }
             }
-
-            if (!hasPartitionLimit) {
-                return filter;
+            if (bottomConjuncts.isEmpty()) {
+                return null;
             }
 
-            Optional<Plan> newWindow = window.pushPartitionLimitThroughWindow(partitionLimit, false);
-            if (!newWindow.isPresent()) {
-                return filter;
-            }
-            return filter.withChildren(newWindow.get());
+            LogicalFilter bottomFilter = new LogicalFilter(bottomConjuncts, window.child());
```

Comment:
do not use Raw class
```suggestion
            LogicalFilter<Plan> bottomFilter = new LogicalFilter<>(bottomConjuncts, window.child());
```